compiler/src/ammtoaga/index.ts

```diff
@@ -493,8 +492,17 @@ const loadStatements = (
       }
     } else if (statement.has('assignments')) {
       const asgn = statement.get('assignments');
-      const resultAddress = localMem[asgn.get('decname').t.trim()];
-      localMemToLine[resultAddress] = line;
+      const asgnName = asgn.get('decname').t.trim();
+      let resultAddress;
+      if (localMem[asgnName]) {
+        resultAddress = localMem[asgnName];
+      } else if (hasClosureArgs && fnArgs.indexOf(asgnName) > -1) {
+        resultAddress = globalMem[asgnName + fnName];
+        mutatedArgs[asgnName + fnName] = true;
+      } else {
```

Comment:
Allows mutations to input arguments, and also flags it as an input arg that is mutated

Reply:
i think a future task will be to use amm as an SSA form and performing variable assignment optimization. it should have a sizeable impact on runtime, as it avoids new `Vec` allocations on each instruction in the avm. This will require the type information to be parsed at the amm level, but the logic can be shared between all `ammto*` stages - it should even result in higher performance with JSR until V8 JIT kicks in